app/services/job_application_pdf_generator.rb

```diff
@@ -1,32 +1,78 @@
 class JobApplicationPdfGenerator
-  include PdfHelper
+  include Prawn::View
+  include PdfUiHelper
 
   def initialize(job_application, vacancy)
-    @job_application = job_application
-    @vacancy = vacancy
+    @datasource = JobApplicationPdf.new(job_application, vacancy)
```

Comment:
Nitpick suggestion :nail_care: 
```suggestion
    @data_source = JobApplicationPdf.new(job_application, vacancy)
```